File: csharp-service-variant-x/Models/Spu.cs
using System.ComponentModel.DataAnnotations;

namespace FlashSale.Api.Models;

/// <summary>
/// Standard Product Unit - represents a product concept (like Saleor's Product)
/// </summary>
public class Spu : BaseEntity
{
    [Required]
    [MaxLength(250)]
    public string Name { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(255)]
    public string Slug { get; set; } = string.Empty;
    
    public string? Description { get; set; }
    
    public bool IsActive { get; set; } = true;
    
    // Navigation properties
    public virtual ICollection<Sku> Skus { get; set; } = new List<Sku>();
    
    public override string ToString() => Name;
}

File: variant-a/java-service/src/main/java/com/flashsale/api/entity/Spu.java
package com.flashsale.api.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.util.ArrayList;
import java.util.List;

/**
 * Standard Product Unit - represents a product concept (like Saleor's Product)
 */
@Entity
@Table(name = "spus", indexes = {
    @Index(name = "idx_spu_slug", columnList = "slug", unique = true)
})
public class Spu extends BaseEntity {

    @NotBlank
    @Size(max = 250)
    @Column(nullable = false)
    private String name;

    @NotBlank
    @Size(max = 255)
    @Column(nullable = false, unique = true)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Boolean isActive = true;

    @OneToMany(mappedBy = "spu", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Sku> skus = new ArrayList<>();

    // Constructors
    public Spu() {}

    public Spu(String name, String slug) {
        this.name = name;
        this.slug = slug;
    }

    // Getters and Setters
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSlug() {
        return slug;
    }

    public void setSlug(String slug) {
        this.slug = slug;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Boolean getIsActive() {
        return isActive;
    }

    public void setIsActive(Boolean isActive) {
        this.isActive = isActive;
    }

    public List<Sku> getSkus() {
        return skus;
    }

    public void setSkus(List<Sku> skus) {
        this.skus = skus;
    }

    @Override
    public String toString() {
        return name;
    }
}
